docs/how-to-source-git.md

```diff
@@ -0,0 +1,180 @@
+```
+$ mkdir systemd-source-git
+$ cd systemd-source-git
+$ git init .
+Initialized empty Git repository in /home/tt/t/systemd-source-git/.git/
+```
+
+We'll add fedora and upstream remotes, and then we'll fetch them:
+```
+$ git remote add upstream https://github.com/systemd/systemd.git
+$ git remote add fedora ssh://ttomecek@pkgs.fedoraproject.org/rpms/systemd.git
+$ git fetch -q upstream
+$ git fetch -q fedora
+```
+
+Systemd also has a dedicated repository with backports to the older releases.
+This is how systemd is packaged in Fedora.
+```
+$ git remote add upstream-stable https://github.com/systemd/systemd-stable.git
+$ git fetch -q upstream-stable
+```
+
+### We can start now
+
+Usually you would try to figure which upstream release is used so that you know
+what to choose for the base. But systemd in Fedora is not using the pristine
+upstream archives.
+
+When we open the Fedora 29 systemd spec file, we can see that upstream uses
+commit hash `8bca4621fc003a148c70248c55aa877dfe61fd3f` for the upstream tarball
+from the systemd-stable repo. We'll start a new git branch in our repo named
+`239-sg` (sg as source-git):
+```
+$ git checkout -B 239-sg 8bca4621fc003a148c70248c55aa877dfe61fd3f
+Switched to a new branch '239-sg'
+```
+
+Right now we have the upstream history which lead to the 239 release.
+Alternatively we could just unpack the upstream tarball and have the history in
+a single commit.
+
+In this case, the upstream history is not marked in a any way, so let's tag it:
+```
+$ git tag upstream-239 HEAD
+```
+Packit will be able to distinguish between additional source-git
+content and upstream history.
+
+We can start layering downstream content on top.
+
+
+### Layering downstream content on top of upstream
+
+To not clutter the root directory in the repo, let's create a new directory
+"fedora/" and just copy in there everything from the f29 fedora branch:
+```
+$ mkdir fedora
+$ git checkout fedora/f29 -- .
+$ git status
+On branch 239-sg
+Changes to be committed:
+  (use "git reset HEAD <file>..." to unstage)
+
+        modified:   .gitignore
+        new file:   0001-Revert-journald-periodically-drop-cache-for-all-dead.patch
+        new file:   0998-resolved-create-etc-resolv.conf-symlink-at-runtime.patch
+        new file:   20-grubby.install
+        new file:   20-yama-ptrace.conf
+        new file:   inittab
+        new file:   purge-nobody-user
+        new file:   sources
+        new file:   split-files.py
+        new file:   sysctl.conf.README
+        new file:   systemd-journal-gatewayd.xml
+        new file:   systemd-journal-remote.xml
+        new file:   systemd-udev-trigger-no-reload.conf
+        new file:   systemd-user
+        new file:   systemd.spec
+        new file:   triggers.systemd
+        new file:   yum-protect-systemd.conf
+```
+
+We have a bunch of new files from f29 branch in the root. We'll move them now to the `fedora/` directory:
+```
+$ mv $(git diff --name-only --cached) fedora/
+```
```

Comment:
(I'd shuffle it a bit)

### Layering downstream content on top of upstream

First get files from f29 branch.

```
$ git checkout fedora/f29 -- .
$ git status
On branch 239-sg
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)
        modified:   .gitignore
        new file:   0001-Revert-journald-periodically-drop-cache-for-all-dead.patch
        new file:   0998-resolved-create-etc-resolv.conf-symlink-at-runtime.patch
        new file:   20-grubby.install
        new file:   20-yama-ptrace.conf
        new file:   inittab
        new file:   purge-nobody-user
        new file:   sources
        new file:   split-files.py
        new file:   sysctl.conf.README
        new file:   systemd-journal-gatewayd.xml
        new file:   systemd-journal-remote.xml
        new file:   systemd-udev-trigger-no-reload.conf
        new file:   systemd-user
        new file:   systemd.spec
        new file:   triggers.systemd
        new file:   yum-protect-systemd.conf
```

We have a bunch of new files from f29 branch in the root.
To not clutter the root directory in the repo, let's create a new directory
"fedora/" and just move in there everything we got from the f29 fedora branch:

```
$ mkdir fedora
$ mv $(git diff --name-only --cached) fedora/
```

Reply:
good suggestion, I updated